tests/unit/test_req_file.py

```diff
@@ -281,7 +281,10 @@ def stub_parse_requirements(
         assert list(process_line(line, "filename", 1)) == [(req, True)]
 
     def test_options_on_a_requirement_line(self):
-        line = "SomeProject --install-option=yo1 --install-option yo2 " '--global-option="yo3" --global-option "yo4"'
+        line = (
+            "SomeProject --install-option=yo1 --install-option yo2 "
```

Comment:
`" ".join([<list of strings>])` would make both black happy and become easier to read (since black will "explode" the list to be one item per line).

Reply:
I'm not keen on doing a runtime join just because we don't like the way black formats things. Either we accept what it does, or we don't use it, IMO. (I'm inclined towards "don't use black" personally, but I'm not going to make a fight out of it).

Reply:
I formatted the strings that way not black FWIW. It's too long to fit on one line, so you have to break it over multiple lines. The current formatting (before this PR) looks like:

```python
        line = 'SomeProject --install-option=yo1 --install-option yo2 '\
               '--global-option="yo3" --global-option "yo4"'
```

Black removed the ``\`` because it considers the backslash character to be bad formatting, but it did *not* collapse the string into a single line (it kept them as two separate strings, because combining them into one string would change the AST, and a hard rule of black currently is that the AST compares the same here).

The use of ``\`` to split this string over multiple lines is an outlier in this code base, generally we use parenthesis and implicit string concatenation throughout pip, so I just adjusted it to use that instead of ``\``.

Reply:
I also don't think a runtime join is the right pattern here FWIW, parenthesis and implicit string concatenation is the de facto mechanism for breaking long strings.